plugins/aws/ec2/ebsSnapshotLifecycle.js

```diff
@@ -0,0 +1,156 @@
+                helpers.addResult(results, 3,
+                    `Unable to EC2 instances: ${helpers.addError(describeInstances)}`, region);
+                return rcb();
+            }
+
+            var getLifecyclePolicies = helpers.addSource(cache, source,
+                ['dlm', 'getLifecyclePolicies', region]);
+
+            if (!getLifecyclePolicies || getLifecyclePolicies.err | !getLifecyclePolicies.data) {
+                helpers.addResult(results, 3,
+                    `Unable to query DLM lifecycle policies: ${helpers.addError(getLifecyclePolicies)}`, region);
+                return rcb();
+            }
+
+            var passingVolumes = [];
+            var passingVolumeTags = {};
+            var passingInstanceTags = {};
+            getLifecyclePolicies.data.forEach(policy => {
+                var getLifecyclePolicy = helpers.addSource(cache, source,
+                    ['dlm', 'getLifecyclePolicy', region, policy.PolicyId]);
+
+                if (!getLifecyclePolicy || getLifecyclePolicy.err || !getLifecyclePolicy.data || !getLifecyclePolicy.data.Policy) {
+                    helpers.addResult(results, 3,
+                        `Unable to query lifecycle policy: ${helpers.addError(getLifecyclePolicy)}`,
+                        region, policy.PolicyId);
+                    return;
+                }
+
+                if (getLifecyclePolicy.data.Policy.State &&
+                    getLifecyclePolicy.data.Policy.State === 'ENABLED' &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails &&
+                    getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags) {
+                    if (getLifecyclePolicy.data.Policy.PolicyDetails.ResourceTypes) {
+                        if (getLifecyclePolicy.data.Policy.PolicyDetails.ResourceTypes.includes('VOLUME')) {
+                            getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags.forEach(tag => {
+                                if (tag.Key && tag.Value) {
+                                    if (passingVolumeTags[tag.Key]) passingVolumeTags[tag.Key].push(tag.Value); 
+                                    else passingVolumeTags[tag.Key] = [tag.Value];
+                                }
+                            });
+                        }
+
+                        if (getLifecyclePolicy.data.Policy.PolicyDetails.ResourceTypes.includes('INSTANCE')) {
+                            getLifecyclePolicy.data.Policy.PolicyDetails.TargetTags.forEach(tag => {
+                                if (tag.Key && tag.Value) {
+                                    if (passingInstanceTags[tag.Key]) passingInstanceTags[tag.Key].push(tag.Value); 
+                                    else passingInstanceTags[tag.Key] = [tag.Value];
+                                }
+                            });
+                        }
+                    }
+                }
+            });
+
+            describeInstances.data.forEach(instance => {
```

Comment:
Dont we need to check describeInstances.data.length as we are doing for describeVolumes.data.length?